asf_heat_pump_suitability/analysis/evaluate_hp_suitability_outputs/pandera_eval_hps_outputs.py

```diff
@@ -0,0 +1,174 @@
+"""
+This script performs data quality checks and statistical analysis on heat pump suitability data.
+
+Key functionalities:
+1. Validates the dataset schema using Pandera (Polars backend).
+2. Detects outliers in numeric columns based on z-scores.
+3. Generates summary statistics for numeric columns.
+4. Logs results and saves a detailed numeric summary to a CSV file.
+"""
+
+from pathlib import Path
+from datetime import datetime
+import logging
+import polars as pl
+import pandera.polars as pa
+import json
+import config.dq_config as cfg
+from pandera.polars import Check, Column, DataFrameSchema
+from asf_heat_pump_suitability.analysis.hn_zones.hnz_utils.log_utils import (
+    setup_logging_and_file_path,
+)
+
+# ------------------------------------------------------------------------------
+# Logging setup
+# ------------------------------------------------------------------------------
+
+setup_logging_and_file_path(output_dir=cfg.OUTPUT_DIR, log_filename=cfg.LOG_FILENAME)
+
+
+# ---------------------------------------------------------------------------
+# 1.  BUILD SCHEMA
+# ---------------------------------------------------------------------------
+
+schema_columns: dict[str, Column] = {}
+
+
+def add(col: str, dtype, *checks, **kw):
+    """
+    Registers or updates a column in the `schema_columns` dictionary for schema validation.
+
+    Args:
+        col (str): The name of the column to register or update.
+        dtype: The data type of the column (e.g., `float`, `str`, `pl.Boolean`).
+        *checks: Any number of Pandera `Check` objects to validate column values.
+        **kw: Additional keyword arguments for column properties (e.g., `required`, `nullable`, `unique`).
+    """
+    if col in schema_columns:  # merge with existing
+        prev = schema_columns[col]
+        checks = tuple(prev.checks) + tuple(checks)  # keep earlier checks
+        kw = {**{k: getattr(prev, k) for k in ("required", "nullable", "unique")}, **kw}
+    schema_columns[col] = Column(dtype, checks=list(checks), **kw)
+
+
+# -- primary key ------------------------------------------------------------ #
+add("lsoa", str, unique=True, required=True, nullable=False)
```

Comment:
```suggestion
add(col="lsoa", dtype=str, unique=True, required=True, nullable=False)
```
I recommend adding key words to the args so it's more readable :) same below

Reply:
I've done this for some of them, however when I do say 'Check.in_range(0, 1)' it has to be positional and keyword arguments cannot appear before it (e.g. col=c)